Northstar.Client/mod/scripts/vscripts/ui/menu_lobby.nut

```diff
@@ -226,6 +229,44 @@ void function InitLobbyMenu()
 	RegisterSignal( "LeaveParty" )
 }
 
+bool function ProgressionIsEnabled()
+{
+	return Progression_GetPreference()
+}
+bool function ProgressionIsDisabled()
+{
+	return !Progression_GetPreference()
+}
+
+void function EnableProgression( var button )
+{
+	Progression_SetPreference(true)
+	UpdateCachedLoadouts_Delayed()
+	DialogData dialogData
+	dialogData.menu = GetMenu( "AnnouncementDialog" )
+	dialogData.header = "#PROGRESSION_ENABLED_HEADER"
+	dialogData.message = "#PROGRESSION_ENABLED_BODY"
+	dialogData.image = $"ui/menu/common/dialog_announcement_1"
+
+	AddDialogButton( dialogData, "#OK" )
+
+	OpenDialog( dialogData )
+}
+
+void function DisableProgression( var button )
+{
+	Progression_SetPreference(false)
+	DialogData dialogData
+	dialogData.menu = GetMenu( "AnnouncementDialog" )
+	dialogData.header = "#PROGRESSION_DISABLED_HEADER"
+	dialogData.message = "#PROGRESSION_DISABLED_BODY"
+	dialogData.image = $"ui/menu/common/dialog_announcement_1"
+
+	AddDialogButton( dialogData, "#OK" )
+	
+	OpenDialog( dialogData )
+}
```

Comment:
I'd prefer it if these would be the same
```squirrel
void function ToggleProgressionEnabled( var button )
{
	bool progressionEnabled = ProgressionIsEnabled()
	Progression_SetPreference(!progressionEnabled)
	DialogData dialogData
	dialogData.menu = GetMenu( "AnnouncementDialog" )
	dialogData.header = progressionEnabled ? "#PROGRESSION_DISABLED_HEADER" : "#PROGRESSION_ENABLED_HEADER"
	dialogData.message = progressionEnabled ? "#PROGRESSION_DISABLED_BODY" : "#PROGRESSION_ENABLED_BODY"
	dialogData.image = $"ui/menu/common/dialog_announcement_1"

	AddDialogButton( dialogData, "#OK" )

	OpenDialog( dialogData )
}
```

Reply:
That relies on the progression preference being sent first (which wasn't always the case) i could change to this though, sure

Reply:
Ok so regarding the button being a toggle, I don't think its a great idea for a couple of reasons:
- A button saying "toggle progression" doesn't tell the user it's current state
- Changing the text of the footer button is far too much effort

Reply:
Why not just a toggle footer that opens a dialog that tells you something like "Progression is currently **enabled**. Do you want to disable progression?"

Reply:
Yeah that would work, I'll do that maybe